src/generator/common/helpers.ts

```diff
@@ -135,15 +138,14 @@ export function formatParamInfoTypeName(param: ParamInfo): string {
 
 // aggregates the Parameter in op.parameters and the first request
 export function aggregateParameters(op: Operation): Array<Parameter> {
-  if (op.requests!.length > 1) {
-    throw console.error('multiple requests NYI');
-  }
   let params = new Array<Parameter>();
   if (op.parameters) {
     params = params.concat(op.parameters);
   }
-  if (op.requests![0].parameters) {
-    params = params.concat(op.requests![0].parameters);
+  for (const req of values(op.requests)) {
```

Comment:
I don't think this is what we want to do.  E.g. multiple requests means that the endpoint could take a JSON object or a file stream.  Some languages will convert this into two separate APIs (`PutWidgetWithBody()` and `PutWidgetWithFileStream()`); others will fold the distinct params into a struct (`PutWidget(payload PutWidgetPayload)`), then will send whatever field is set in the payaload.  I think we need to discuss this more in depth to decide which approach we want to take.

Reply:
Yes, you're right. I'll remove that and leave it as it was before for now.

Reply:
Actually I'll throw an error if we receive multiple responses so we know

Reply:
Removed